### DISCUSSION TITLE
token()

### DISCUSSION THREAD

**Discussion by vzaliva** [tree-sitter/tree-sitter#2873] [Q&A]
I have a working grammar which has a rule like:
```
foo: $ => choice("a","b","c")
```

I do not need individual tokens so I tried to change it to:
```
foo: $ => token(choice("a","b","c"))
```

Which broke my parsing. My understanding is that it should not have changed how stuff is parsed, just what values are returned in the resulting AST. Could someone please shed some light on this? Thanks!

**Comment by amaanq** [comment]
Token *can* change how parsing happens, but I'd need a bit more context than what you've provided to be able to assist/debug your issue, a minimal grammar + source file to parse that can reproduce your issue would be great!
